src/Planners/tests/ContactListTest.cpp

```diff
@@ -1,28 +1,26 @@
 /**
  * @file ContactListTest.cpp
- * @authors Stefano Dafarra
+ * @authors Stefano Dafarra, Giulio Romualdi
  * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
  * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
  */
 
 // Catch2
 #include <catch2/catch.hpp>
 
-#include <iDynTree/Core/Utils.h>
-#include <iDynTree/Core/EigenHelpers.h>
+// manif
+#include <manif/manif.h>
 
 #include <BipedalLocomotion/Planners/ContactList.h>
 
 using namespace BipedalLocomotion::Planners;
 
 bool contactsAreEqual(const Contact& c1, const Contact& c2)
 {
-    return (c1.type == c2.type) &&
-           (c1.name == c2.name) &&
-           (iDynTree::toEigen((c1.pose * c2.pose.inverse()).asHomogeneousTransform()).isApprox(
-               iDynTree::toEigen(iDynTree::Transform::Identity().asHomogeneousTransform()))) &&
-           (iDynTree::checkDoublesAreEqual(c1.activationTime, c2.activationTime)) &&
-           (iDynTree::checkDoublesAreEqual(c1.deactivationTime, c2.deactivationTime));
+    return (c1.type == c2.type) && (c1.name == c2.name)
+           && (((c1.pose * c2.pose.inverse()).log()).isApprox(Eigen::Matrix<double, 6, 1>::Zero()))
+           && (c1.activationTime == c2.activationTime)
+           && (c1.deactivationTime == c2.deactivationTime);
```

Comment:
is this a strong check for doubles? Wouldn't you need a check with tolerance as it was offered by iDynTree::checkDoublesAreEqual(a, b)?

Reply:
Initially, this was to check if two contacts are *exactly* the same, i.e. two copies of the same object. We could try to have some coherence also with the transformations check. Either they are both checked precisely or with a tolerance.

Reply:
https://github.com/dic-iit/bipedal-locomotion-framework/pull/82/commits/fba1c02cef442dee58e39f09f9eb1aa75cbe2995 goes in the direction of what @prashanthr05 suggested in https://github.com/dic-iit/bipedal-locomotion-framework/pull/82#discussion_r470568742